src/flatc.cpp

```diff
@@ -1018,12 +1025,29 @@ int FlatCompiler::Compile(const FlatCOptions &options) {
 }
 
 bool FlatCompiler::RegisterCodeGenerator(
-    const std::string &flag, std::shared_ptr<CodeGenerator> code_generator) {
-  if (code_generators_.find(flag) != code_generators_.end()) {
-    Error("multiple generators registered under: " + flag, false, false);
+    const FlatCOption &option, std::shared_ptr<CodeGenerator> code_generator) {
+  if (!option.short_opt.empty() &&
+      code_generators_.find(option.short_opt) != code_generators_.end()) {
+    Error("multiple generators registered under: -" + option.short_opt, false,
+          false);
     return false;
+  } else {
+    std::string short_opt_flag = "-" + option.short_opt;
+    code_generators_[short_opt_flag] = code_generator;
   }
-  code_generators_[flag] = std::move(code_generator);
+
+  if (!option.long_opt.empty() &&
+      code_generators_.find(option.long_opt) != code_generators_.end()) {
+    Error("multiple generators registered under: --" + option.long_opt, false,
+          false);
+    return false;
+  } else {
+    std::string long_opt_flag = "--" + option.long_opt;
+    code_generators_[long_opt_flag] = code_generator;
+  }
+
+  AppendShortOption(short_flag_help_usage, option);
```

Comment:
So this will make the help text dependent on registration order. I wonder if its better to just store them until the help text generation is requested, then we can sort (in some order) and build the string when needed?